File: kide-enterprise/backend/app/schemas/ingestion.py
from typing import Any, Dict, List, Optional
from datetime import datetime
from pydantic import BaseModel, Field

class IngestionJobResponse(BaseModel):
    id: int
    job_uuid: str
    org_id: int
    user_id: Optional[int] = None
    filename: str
    file_type: str
    file_size_bytes: int
    status: str
    progress_pct: int
    stage_message: str
    extracted_devices_count: int
    extracted_capabilities_count: int
    extracted_operations_count: int
    extracted_parameters_count: int
    error_message: Optional[str] = None
    created_at: datetime
    completed_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class IngestionJobCreate(BaseModel):
    filename: str
    file_type: str = "txt"
    content: str

class SampleIngestRequest(BaseModel):
    sample_id: str

class StagedArtifactResponse(BaseModel):
    id: int
    artifact_uuid: str
    job_id: int
    org_id: int
    name: str
    category: str
    protocol: str
    status: str
    device_metadata: Dict[str, Any]
    extracted_datapoints: List[Dict[str, Any]]
    extracted_commands: List[Dict[str, Any]]
    extracted_alarms: List[Dict[str, Any]]
    extracted_capabilities: List[Dict[str, Any]]
    generated_dml: str
    generated_mnc: str
    generated_cap: str
    generated_op: str
    review_notes: Optional[str] = None
    reviewed_by_user_id: Optional[int] = None
    reviewed_at: Optional[datetime] = None
    created_at: datetime

    class Config:
        from_attributes = True

class ArtifactReviewActionRequest(BaseModel):
    notes: Optional[str] = None

class NotificationResponse(BaseModel):
    id: int
    org_id: Optional[int] = None
    user_id: Optional[int] = None
    recipient_email: str
    notification_type: str
    subject: str
    body_text: str
    body_html: Optional[str] = None
    status: str
    channel: str
    metadata_json: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True

class SampleTemplate(BaseModel):
    id: str
    name: str
    category: str
    protocol: str
    manufacturer: str
    description: str
    file_type: str

File: kide-enterprise/frontend/src/types/ingestion.ts
export interface IngestionJob {
  id: number;
  job_uuid: string;
  org_id: number;
  user_id?: number;
  filename: string;
  file_type: string;
  file_size_bytes: number;
  status: 'pending' | 'processing' | 'completed' | 'failed';
  progress_pct: number;
  stage_message: string;
  extracted_devices_count: number;
  extracted_capabilities_count: number;
  extracted_operations_count: number;
  extracted_parameters_count: number;
  error_message?: string;
  created_at: string;
  completed_at?: string;
}

export interface ExtractedDatapoint {
  name: string;
  type: string;
  unit: string;
  min_val: number;
  max_val: number;
  description: string;
}

export interface ExtractedCommand {
  name: string;
  parameters: any[];
  return_type: string;
  description: string;
}

export interface ExtractedAlarm {
  name: string;
  level: number;
  condition?: string;
  description: string;
}

export interface StagedKnowledgeArtifact {
  id: number;
  artifact_uuid: string;
  job_id: number;
  org_id: number;
  name: string;
  category: string;
  protocol: string;
  status: 'draft' | 'approved' | 'rejected' | 'promoted';
  device_metadata: Record<string, any>;
  extracted_datapoints: ExtractedDatapoint[];
  extracted_commands: ExtractedCommand[];
  extracted_alarms: ExtractedAlarm[];
  extracted_capabilities: any[];
  generated_dml: string;
  generated_mnc: string;
  generated_cap: string;
  generated_op: string;
  review_notes?: string;
  reviewed_by_user_id?: number;
  reviewed_at?: string;
  created_at: string;
}

export interface NotificationItem {
  id: number;
  org_id?: number;
  user_id?: number;
  recipient_email: string;
  notification_type: string;
  subject: string;
  body_text: string;
  body_html?: string;
  status: string;
  channel: string;
  metadata_json?: string;
  created_at: string;
}

export interface NotificationStats {
  total_notifications: number;
  sent_count: number;
  failed_count: number;
  latest_notification?: string;
}

export interface SampleTemplate {
  id: string;
  name: string;
  category: string;
  protocol: string;
  manufacturer: string;
  description: string;
  file_type: string;
}

